vignettes/testing_demo.Rmd

```diff
@@ -1,51 +1,467 @@
+        geom_tile() +
+        scale_fill_brewer(palette = "Dark2", labels = c("Passed", "Failed"), 
+                          na.value = "darkgrey", name = "Quality flag") +
+        labs(x = "Country", y = "Date") +
+        facet_grid(data_source ~ who_region, scales = "free_x", space = "free_x") +
+        theme(axis.text.x = element_text(angle = 45, size = 8, hjust = 1))
+  )
+```
+
+## Checking functions to make sure they return the data even when OWID switches off
+
+Checking functions to make sure they return data even when OWID data stops being
+updated:
+```{r}
+# same as function in pkg except passing through testing_long
+get_testing_mod <- function (testing_long,
+                             analysis_date = Sys.Date() - 1L) {
+    preferred <- get_preferred_testpos7(testing_long, last_X_days = 14, 
+        analysis_date = analysis_date)
+    preferred_long <- testing_long %>% left_join(preferred %>% 
+        select(id, preferred_source), by = "id") %>% filter(data_source == 
+        preferred_source)
+    preferred_long_locf <- 
+      preferred_long %>%
+        group_by(id) %>%
+        arrange(date) %>%
+        mutate(new_tests_smoothed_per_thousand = zoo::na.locf(new_tests_daily7_per_1k,
+          na.rm = F, maxgap = 14
+        ), new_tests_smoothed_per_thousand_14 = zoo::na.locf(new_tests_daily14_per_1k,
+          na.rm = F, maxgap = 14
+        ), positive_rate = zoo::na.locf(positive_rate_7day,
+          na.rm = F, maxgap = 14
+        )) %>%
+        ungroup(.) %>%
+        select(
+          id,
+          date, new_tests_smoothed_per_thousand, new_tests_smoothed_per_thousand_14,
+          positive_rate, data_source
+        )
+    return(preferred_long_locf)
+}
+
+# seems to work with just the 
+check_test <-
+  testing_long %>%
+  filter(data_source == "FIND" | (data_source == "OWID" & date <= "2022-06-01")) %>%
+  get_testing_mod() 
+
+check_test %>%
+  ggplot(aes(x = date, y = id, fill = new_tests_smoothed_per_thousand)) +
+  geom_tile()
+
+check_test %>%
+  ggplot(aes(x = date, y = id, fill = data_source)) +
+  geom_tile()
+```
+
+
+And with other sequence of functions:
+```{r}
+preferred_source <- 
+  get_preferred_testpos7(
+    testing_long %>%
+    filter(data_source == "FIND" | (data_source == "OWID" & date <= "2022-06-01"))
+  )
+preferred_source
+```
+
+Or just using test numbers over the past 14 days:
+```{r}
+get_preferred_tests14(
+  testing_long %>%
+    filter(data_source == "FIND" | (data_source == "OWID" & date <= "2022-06-01"))
+)
+```
+
+```{r}
+testing_long %>%
+  filter(data_source == "FIND" | (data_source == "OWID" & date <= "2022-06-01")) %>%
+  left_join(preferred_source %>% 
+            select(id, preferred_source), by = "id") %>%
+  filter(data_source == preferred_source)
+```
+
+
+## Summary of testing data processing
+
+- For each data source, we do the following:
+
+    + OWID linearly interpolates the cumulative number of tests to date and
```

Comment:
I think this is the main part that could use a check. Bhezad's documentation pointed to the github link below where it shows that as of Jan, OWID actually removed the code where interpolation doesn't happen if dates are more than 31 days apart? I think? I don't know how much this affects the data, but I'm guessing not much of an issue given the switch to FIND. And also could be wrong, so feel free to correct if you have a better handle on it!

Reply:
I think this doesn't impact us anymore since we also linearly interpolate both using `zoo::na.approx` in `get_testing_long()` so they're performed in the same way.

Reply:
oh perfect! I will simplify this section then and push up the changes this afternoon!